crates/symbolicator-proguard/src/symbolication.rs

```diff
@@ -184,10 +185,25 @@ impl ProguardService {
         frame: &JvmFrame,
         release_package: Option<&str>,
     ) -> Vec<JvmFrame> {
-        // First, try to remap the whole frame. This only works if it has a line number.
-        if let Some(lineno) = frame.lineno {
-            let proguard_frame =
-                proguard::StackFrame::new(&frame.module, &frame.function, lineno as usize);
+        let stack_frame = if let Some(lineno) = frame.lineno {
+            Some(proguard::StackFrame::new(
+                &frame.module,
+                &frame.function,
+                lineno as usize,
+            ))
+        } else {
+            frame.parameters.as_ref().map(|params| {
+                proguard::StackFrame::with_parameters(
+                    &frame.module,
+                    &frame.function,
+                    params.as_str(),
+                )
+            })
+        };
```

Comment:
This can be written as
```suggestion
        let stack_frame = frame
            .lineno
            .map(|lineno| {
                proguard::StackFrame::new(&frame.module, &frame.function, lineno as usize)
            })
            .or_else(|| {
                frame.parameters.as_ref().map(|params| {
                    proguard::StackFrame::with_parameters(
                        &frame.module,
                        &frame.function,
                        params.as_str(),
                    )
                })
            });
```
which I believe is slightly nicer. I'll leave it up to you though.